Dockerfile

```diff
@@ -2,34 +2,62 @@ FROM azul/zulu-openjdk-debian:11.0.11
 
 WORKDIR /opt
 
-ENV HADOOP_HOME=/opt/hadoop-2.10.1
-ENV HIVE_HOME=/opt/apache-hive-2.3.9-bin
-# Include additional jars
-ENV HADOOP_CLASSPATH=/opt/hadoop-2.10.1/share/hadoop/tools/lib/aws-java-sdk-bundle-1.11.271.jar:/opt/hadoop-2.10.1/share/hadoop/tools/lib/hadoop-aws-2.10.1.jar
-
-RUN apt-get update && \
+ENV HADOOP_HOME=/opt/hadoop
+ENV HADOOP_VERSION=3.3.1
+ENV HIVE_HOME=/opt/hive
+ENV HIVE_VERSION=3.1.2
+
+RUN mkdir ${HIVE_HOME}
+RUN mkdir ${HADOOP_HOME}
+RUN apt-get clean && \
+    apt-get update && \
     apt-get upgrade -y && \
     apt-get -qqy install curl && \
-    curl -L https://dlcdn.apache.org/hive/hive-2.3.9/apache-hive-2.3.9-bin.tar.gz | tar zxf - && \
-    curl -L https://dlcdn.apache.org/hadoop/common/hadoop-2.10.1/hadoop-2.10.1.tar.gz | tar zxf - && \
+    curl -L https://dlcdn.apache.org/hive/hive-${HIVE_VERSION}/apache-hive-${HIVE_VERSION}-bin.tar.gz | tar zxf - && \
+    curl -L https://dlcdn.apache.org/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz | tar zxf - && \
+    mv apache-hive-${HIVE_VERSION}-bin/* ${HIVE_HOME} && \
+    mv hadoop-${HADOOP_VERSION}/* ${HADOOP_HOME} && \
     apt-get install --only-upgrade openssl libssl1.1 && \
     apt-get install -y libk5crypto3 libkrb5-3 libsqlite3-0 zip
 
 RUN rm ${HIVE_HOME}/lib/postgresql-9.4.1208.jre7.jar
 
 RUN curl -o ${HIVE_HOME}/lib/postgresql-9.4.1212.jre7.jar -L https://jdbc.postgresql.org/download/postgresql-9.4.1212.jre7.jar
 
-COPY conf ${HIVE_HOME}/conf
-
-RUN groupadd -r hive --gid=1000 && \
-    useradd -r -g hive --uid=1000 -d ${HIVE_HOME} hive && \
-    chown hive:hive -R ${HIVE_HOME}
+# Configure Hadoop AWS Jars to be available to hive
+RUN ln -s ${HADOOP_HOME}/share/hadoop/tools/lib/*aws* ${HIVE_HOME}/lib
 
-RUN zip -q -d /opt/apache-hive-*-bin/lib/log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class
-
-USER hive
+COPY conf ${HIVE_HOME}/conf
+COPY scripts/entrypoint.sh ${HIVE_HOME}/entrypoint.sh
+
+RUN zip -q -d ${HIVE_HOME}/lib/log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class
```

Comment:
do we still need this?

Reply:
No, that was a temporary patch until we could upgrade. If you run it now it errors with "nothing to do".

Reply:
I don't think it hurts to leave it in.

Reply:
But technically no, it's ultimately not doing anything.

Reply:
Everything in `${HIVE_HOME}/lib/log4j-*` ends up getting deleted in the next step before the latest log4j is installed.

Reply:
Let me know if you'd like me to remove it.

Reply:
Well my point more here is, the docker file is hard to parse as reviewer/user who wants to contribute later point in time.. Suggestion would be to give additional context of what's happening as a comment above each block..

If this is not need then would prefer having it removed/cleaned up..

Reply:
Addressed